cfgov/scripts/enforcement_transfer.py

```diff
@@ -0,0 +1,165 @@
+
+        tags = page.tags.all()
+        sidefoot = get_stream_data(page, 'sidefoot')
+        sidefoot_fields = []
+        continue_outer_loop = False
+
+        court = ''
+        docket_number = None
+        status = None
+        institution_type = None
+        date_filed = None
+
+        for field in sidefoot:
+            if field['type'] == 'related_metadata':
+                metadata = field['value']['content']
+                for item in metadata:
+                    heading = item['value']['heading']
+
+                    try:
+                        if heading == 'Court':
+                            court = get_metadata_value(item, heading)
+                        elif heading in ['Institution type', 'Institution']:
+                            institution_type = get_metadata_value(
+                                item,
+                                heading
+                            )
+                        elif heading == 'Status':
+                            listed_status = clean_status(
+                                get_metadata_value(item, heading)
+                            )
+                            if listed_status not in STATUS_OPTIONS:
+                                status = STATUS_OPTIONS[0]
+                            else:
+                                status = listed_status
+                        elif heading.lower() in [
+                            'docket number',
+                            'case number',
+                            'case numbers',
+                            'file number',
+                            'file numbers',
+                            'civil action number'
+                        ]:
+                            docket_number = item['value']['blob']
+                        elif heading == 'Date filed':
+                            date_filed = item['value']['date']
+                        elif (
+                            heading == 'Category' or
+                            heading == 'Topics' or
+                            heading == 'Last updated'
+                        ):
+                            continue
+                    except Exception:
+                        errors.append(
+                            f'Error accessing {heading} metadata for {url}'
+                        )
+                        continue_outer_loop = True
+            else:
+                sidefoot_fields.append(field)
+
+        if continue_outer_loop:
+            continue
+        if not institution_type:
+            errors.append(f'Missing Institution type for {url}')
+            continue
+        if not status:
+            errors.append(f'Missing Status for {url}')
+            continue
+        if not docket_number:
+            errors.append(f'Missing Docket number for {url}')
+            continue
+
+        statuses = [EnforcementActionStatus(status=status)]
+
+        page.delete()
```

Comment:
One risk of deleting the existing pages like this is that any links to them _based on their primary key_ will break, because the primary key of the new EAPs won't be the same as the old DDPs. This includes links in rich text fields or any other StreamField block that uses the Wagtail page chooser to make the link. Is that something you're concerned about with this change?

Another theoretical risk is that you have to be careful about not deleting any model instances that depend on the page with an `on_delete=models.CASCADE` relationship. For example, any models related to this page via an InlinePanel. I don't think we have any of those, but it's worth making sure.

Deleting these will also delete any child pages, if any exist. Our internal policy has been to disable page deletion entirely. Might it be better to instead "Trash" them as per our standard Wagtail deletion process?

(It might be interesting to chat about experimenting with _changing_ the page types of these existing pages. This is something that is definitely not supported in Wagtail, but is at least theoretically possible via some potentially-unwise database hacking. I recently did this [here](https://github.com/cfpb/cfgov-refresh/blob/343a1dcacc9cfd774063e671da67d13577761116/cfgov/v1/migrations/0203_deprecate_spanish_home_page.py#L11) as part of cleaning up the deprecate SpanishHomePage model, but that was made much simpler because of the [specifics](https://github.com/cfpb/cfgov-refresh/blob/343a1dcacc9cfd774063e671da67d13577761116/cfgov/v1/migrations/0203_deprecate_spanish_home_page.py#L32-L38) of the migration I was doing. It'd be great to have a general "change page types" feature to have in Wagtail -- see https://github.com/wagtail/wagtail/issues/4949 -- so I'd love to talk with you more about your needs if that seems like the right tool here.)

Reply:
> Is that something you're concerned about with this change

Not particularly concerned, but there certainly may be examples of that. Ultimately the only reason I was deleting them at all is so I could immediately re-use the slug. However, I imagine moving the initial pages to the Trash would free up the slug for the new EAP so everything would still work.

I did a bit of reading about changing the page type with db hacking and abandoned it as something that was outside my abilities to do comfortably, whereas I'm much more confident verifying that the sharp ends of this script (some of which you've listed above) don't poke too much. I'd say if we can assure ourselves we won't break a ton of links or delete other model instances, we should stick with this approach in this case.